sql-plugin/src/main/scala/org/apache/spark/sql/rapids/datetimeExpressions.scala

```diff
@@ -140,7 +140,11 @@ abstract class GpuTimeMath(
     withResourceIfAllowed(left.columnarEval(batch)) { lhs =>
       withResourceIfAllowed(right.columnarEval(batch)) { rhs =>
         (lhs, rhs) match {
-          case (l: GpuColumnVector, intvl: CalendarInterval) =>
+          case (l: GpuColumnVector, intvlS: GpuScalar)
+              if intvlS.getValue.isInstanceOf[CalendarInterval] =>
```

Comment:
Wouldn't it be cheaper to check `intvlS.dataType`?  That way we don't have to unpack a potentially GPU-side scalar unless it's the type we expect not to be on the GPU.

Reply:
Updated.
I thought it can avoid a null check by using the `value`. But now it still does not need the null check when using the `dataType`.